blockchain/blockhashHandler_test.go

```diff
@@ -0,0 +1,191 @@
+package blockchain
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-sdk-go/data"
+	"github.com/multiversx/mx-sdk-go/testsCommon"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewBlockhashHandler(t *testing.T) {
+	t.Run("invalid polling interval", func(t *testing.T) {
+		args := createMockArgsBlockhashHandler()
+		args.pollingInterval = time.Millisecond
+		handler, err := NewBlockhashHandler(args)
+
+		assert.True(t, check.IfNil(handler))
+		assert.Equal(t, err, fmt.Errorf("%w in checkArgs for value PollingInterval", ErrInvalidValue))
+	})
+	t.Run("invalid block ttl", func(t *testing.T) {
+		args := createMockArgsBlockhashHandler()
+		args.blockTtl = time.Millisecond
+		handler, err := NewBlockhashHandler(args)
+
+		assert.True(t, check.IfNil(handler))
+		assert.Equal(t, err, fmt.Errorf("%w in checkArgs for value BlockTtl", ErrInvalidValue))
+	})
+	t.Run("nil http client wrapper", func(t *testing.T) {
+		args := createMockArgsBlockhashHandler()
+		args.httpClientWrapper = nil
+		handler, err := NewBlockhashHandler(args)
+
+		assert.True(t, check.IfNil(handler))
+		assert.Equal(t, err, ErrNilHTTPClientWrapper)
+	})
+	t.Run("should work", func(t *testing.T) {
+		args := createMockArgsBlockhashHandler()
+		handler, err := NewBlockhashHandler(args)
+
+		assert.False(t, check.IfNil(handler))
+		assert.Nil(t, err)
+	})
+	t.Run("process loop should start", func(t *testing.T) {
+		args := createMockArgsBlockhashHandler()
+		handler, _ := NewBlockhashHandler(args)
+		time.Sleep(2 * time.Second) // wait for the process loop to start
+
+		assert.True(t, handler.loopStatus.IsSet())
+	})
+}
+
+func TestGetBlockTimestampByHash_Success(t *testing.T) {
```

Comment:
for all these tests, perhaps export a new constructor that does not start the processLoop which is not needed